### PR TITLE
PIM-17590: Update to latest mongodb dependencies

### PR DESCRIPTION
**Pull request by MatPayette**
This PR aims to update this fork in order to allow updating other repos that depend on it. It is possible that it is no longer needed with the new mongodb versions, however one should make sure that that is the case before taking any removal action. For now, this PR should suffice.

### CODE REVIEW

**Review by pgxtreme — APPROVED** [review]

File: src/utils.ts [hunk 1]
```diff
@@ -96,7 +96,22 @@ export const normalizeDirectionParams = ({
         return {
             limit: Math.max(1, last ?? 20),
             cursor: before ? decodeCursor(before) : null,
-            sort: mapValues(sort, (value) => value * -1),
+            sort: mapValues(
```
Comment (chris-dallaire) [inline src/utils.ts:99]: Could you ad a comment describing why we reverse these values? Without any explanation, it looks like a bug/typo

File: test/unit/utils.spec.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ describe("buildCursor", () => {
             name: "John Doe",
         };
 
-        const sort = {
+        const sort: Sort = {
             createdAt: 1,
```
Comment (chris-dallaire) [inline test/unit/utils.spec.ts:22]: Question: Can we use `SortDirection` instead of the hardcoded integers?

**Review by chris-dallaire — APPROVED** [review]

File: test/unit/utils.spec.ts (hunk 2 above)
Comment (MatPayette) [reply test/unit/utils.spec.ts:22]: `SortDirection` isn't an enum :p It's a direct union of values such as `1`, `-1`, `asc`, etc.

File: src/utils.ts (hunk 1 above)
Comment (MatPayette) [reply src/utils.ts:99]: I've actually no idea what the logic is tbqh; my goal is merely to make the existing code work with the new mongodb version :)

**Review by Samuel-Langlois-BimOne — APPROVED** [review]

File: src/utils.ts (hunk 1 above)
Comment (chris-dallaire) [reply src/utils.ts:99]: Ah yes sorry I didn't realize that logic was already there. All good!

File: test/unit/utils.spec.ts (hunk 2 above)
Comment (chris-dallaire) [reply test/unit/utils.spec.ts:22]: Makes sense!

**Pull request merged**